dll/win32/shell32/wine/shellpath.c

```diff
@@ -146,90 +146,136 @@ static BOOL WINAPI PathMakeAbsoluteW(LPWSTR path)
 }
 #endif
 
-/* NOTE: GetShortPathName fails if the pathname didn't exist.
-         GetShortPathNameAbsentW should set the short path name that even doesn't exist. */
-static DWORD GetShortPathNameAbsentW(LPCWSTR pszLong, LPWSTR pszShort, DWORD cchShort)
-{
-    FIXME("GetShortPathNameAbsentW(%ls, %p, %ld): stub\n", pszLong, pszShort, cchShort);
-    StringCchCopyW(pszShort, cchShort, pszLong);
-    return lstrlenW(pszShort);
-}
-
-BOOL WINAPI IsLFNDriveW(LPCWSTR lpszPath);
-
-/* @unconfirmed */
+/* @implemented */
 static VOID WINAPI PathQualifyExW(LPWSTR pszPath, LPCWSTR pszDir, DWORD dwFlags)
 {
-    WCHAR szRoot[MAX_PATH], szCopy[MAX_PATH], szCurDir[MAX_PATH];
-    LPWSTR pch;
-    LONG cch;
-    BOOL bCheckLFN;
+    INT iDrive;
+    WCHAR szTemp[MAX_PATH], szRoot[MAX_PATH];
+    LPWSTR pchTemp, pchPath, pchPathEnd;
 
-    if (FAILED(StringCchCopyW(szCopy, _countof(szCopy), pszPath)))
+    /* FIXME: Short pathname */
+
+    /* Copy path to szTemp */
```

Comment:
A good comment explains why such a copy is done (instead of rerpeating what the name of the function called already tells).

Reply:
Improved.